src/main/java/hudson/plugins/promoted_builds/Promotion.java

```diff
@@ -87,6 +90,13 @@ public Status getStatus() {
         return getTarget().getAction(PromotedBuildAction.class).getPromotion(getParent().getName());
     }
 
+	public String getIcon() {
+		return icon;
+	}
+
+	public void setIcon(String icon) {
```

Comment:
With the current plugin, the badge icon can only be configured before-hand in the project configuration page. If I want the promotion job to dynamically determine what type of badge to associate to the promotion as part of the a build step then that can't be done now. I can either have the promotion job succeed in which case the badge configured at the project level gets associated with the promotion, or I can fail the promotion in which case a red-icon is displayed.

What I would like to do with the enhancement is allow one of the build steps in the promotion job to assign a badge while still allowing the promotion job to succeed.

Here is a screenshot of how the scenario I described. When the promotion job runs for the first time I have a groovy build step that sets the badge icon to gold-star. When the promotion job is run the second time to delete the package, I set the badge icon to an empty string in the groovy build step.

![image](https://cloud.githubusercontent.com/assets/1396289/9413915/c1edec6e-47e9-11e5-8cb2-897dc7d6f687.png)